src/main/scala/org/broadinstitute/dsde/firecloud/model/DUOS.scala

```diff
@@ -1,9 +1,110 @@
 package org.broadinstitute.dsde.firecloud.model
 
-import spray.json.JsObject
+import spray.json.{JsObject, JsValue, _}
+import DefaultJsonProtocol._
+
+import scala.util.{Success, Try}
 
 object DUOS {
 
+  case class DuosDataUse(
+    generalUse: Option[Boolean] = None,
+    hmbResearch: Option[Boolean] = None,
+    diseaseRestrictions: Option[Seq[String]] = None,
+    populationOriginsAncestry: Option[Boolean] = None,
+    populationStructure: Option[Boolean] = None,
+    commercialUse: Option[Boolean] = None,
+    methodsResearch: Option[Boolean] = None,
+    aggregateResearch: Option[String] = None,
+    controlSetOption: Option[String] = None,
+    gender: Option[String] = None,
+    pediatric: Option[Boolean] = None,
+    populationRestrictions: Option[Seq[String]] = None,
+    dateRestriction: Option[String] = None,
+    recontactingDataSubjects: Option[Boolean] = None,
+    recontactMay: Option[String] = None,
+    recontactMust: Option[String] = None,
+    genomicPhenotypicData: Option[String] = None,
+    otherRestrictions: Option[Boolean] = None,
+    cloudStorage: Option[String] = None,
+    ethicsApprovalRequired: Option[Boolean] = None,
+    geographicalRestrictions: Option[String] = None,
+    other: Option[String] = None,
+    illegalBehavior: Option[Boolean] = None,
+    addiction: Option[Boolean] = None,
+    sexualDiseases: Option[Boolean] = None,
+    stigmatizeDiseases: Option[Boolean] = None,
+    vulnerablePopulations: Option[Boolean] = None,
+    psychologicalTraits: Option[Boolean] = None,
+    nonBiomedical: Option[Boolean] = None
+  )
+
+  object DuosDataUse {
+    def apply(jsValues: Map[String, JsValue]): DuosDataUse = {
+      def getBoolean(f: String): Option[Boolean] = {
+        jsValues.get(f) match {
+          case x if x.isDefined =>
+            Try(jsValues(f).convertTo[Boolean]) match {
+              case Success(booleanVal) => Some(booleanVal)
+              case _ => None
+            }
+          case _ => None
+        }
+      }
+      def getSeqString(f: String): Option[Seq[String]] = {
+        jsValues.get(f) match {
+          case x if x.isDefined =>
+            Try(jsValues(f).convertTo[JsArray].elements.map(_.convertTo[String])) match {
+              case Success(listVal) => Some(listVal)
+              case _ => None
+            }
+          case _ => None
+        }
+      }
+      def getString(f: String): Option[String] = {
+        jsValues.get(f) match {
+          case x if x.isDefined =>
+            Try(jsValues(f).convertTo[String]) match {
+              case Success(stringVal) => Some(stringVal)
+              case _ => None
+            }
+          case _ => None
+        }
+      }
```

Comment:
I'd look to use `JsObject`'s internal type system to handle all of these. Something like:
```
      def getBoolean(f: String): Option[Boolean] = {
        jsValues.get(f) match {
          case Some(b:JsBoolean) => Some(b.value)
          case _ => None
        }
      }
      def getSeqString(f: String): Option[Seq[String]] = {
        jsValues.get(f) match {
          case Some(a:JsArray) => Some(a.elements.collect {
            case s:JsString => s.value
          })
          case _ => None
        }
      }
      def getString(f: String): Option[String] = {
        jsValues.get(f) match {
          case Some(s:JsString) => Some(s.value)
          case _ => None
        }
      }
```

Reply:
👍  I think I need a combination of this and Kaan's approach.